### PR TITLE
Fixed inference between type placeholders with non-string constraints in template literal types

### PR DESCRIPTION
**Pull request by Andarist**
fixes https://github.com/microsoft/TypeScript/issues/57807

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -26058,6 +26060,16 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     // allowed template literal placeholder types, infer from a literal type corresponding to the constraint.
                     if (source.flags & TypeFlags.StringLiteral && target.flags & TypeFlags.TypeVariable) {
                         const inferenceContext = getInferenceInfoForType(target);
+                        // andarist here???
```
Comment (gabritto) [inline src/compiler/checker.ts:26063]: Did you forget to remove/edit this?

**Comment by gabritto** [comment]
@typescript-bot run DT
@typescript-bot user test this
@typescript-bot test top400
@typescript-bot perf test this

File: src/compiler/checker.ts (hunk 1 above)
Comment (Andarist) [reply src/compiler/checker.ts:26063]: Yes 😅 i’ll remove this embarrassing leftover in a moment 😉

**Review by gabritto — APPROVED** [review]

**Comment by gabritto** [comment]
@typescript-bot run DT

**Comment by gabritto** [comment]
@typescript-bot run dt

**Pull request merged**